### PR TITLE
parametric: use DD_TRACE_SAMPLE_RATE in span sampling tests

### PR DESCRIPTION
**Pull request by ZStriker19**
## Description

Before we used the setting of manual_drop to 1 in order to modulate the tracer's decision to keep or drop the span. However changing manual_drop doesn't effect the Go tracer (and possibly other tracer's) decision when setting the tag `_sampling_priority_v1` which is what  matters when making the decision to try single span sampling or not. Using `DD_TRACE_SAMPLE_RATE` to modulate the tracer's decision is more consistent across tracers and easier to understand.

I did keep the option to add manual_drop in the generate_span method, as it was the only option I know of to modulate trace sampling for Python during runtime which is necessary for sss011.

## Check list

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: parametric/test_span_sampling.py [hunk 1]
```diff
@@ -234,31 +274,42 @@ def test_keep_span_with_stats_computation_sss010(test_agent, test_client: _TestT
 
 
 @pytest.mark.skip_library("dotnet", "Not implemented")
-@pytest.mark.skip_library("golang", "Not implemented")
+@pytest.mark.skip_library("golang", "The Go tracer does not have an easy way to modulate trace sampling once started")
```
Comment (Kyle-Verhoog) [inline parametric/test_span_sampling.py:277]: no easy way, or no way at all?

**Review by Kyle-Verhoog — APPROVED** [review]
I'm not familiar enough with sampling to give a good review on the configuration change but the code lgtm

Maybe worth pulling in @dgoffredo to review as well

File: parametric/test_span_sampling.py (hunk 1 above)
Comment (ZStriker19) [reply parametric/test_span_sampling.py:277]: No way that I'm aware of or that anyone from the Go team offered.

Comment (Kyle-Verhoog) [reply parametric/test_span_sampling.py:277]: maybe worth saying that there's no known way then, else it sounds like we're just lazy

Comment (ZStriker19) [inline parametric/test_span_sampling.py:277]:
```suggestion
@pytest.mark.skip_library("golang", "The Go tracer does not have a way to modulate trace sampling once started")
```

**Review by dgoffredo — APPROVED** [review]
This change highlights the leeway that special tags give to implementations.

Looks good to me.  `DD_TRACE_SAMPLE_RATE={0,1}` has the same effect as `SetTag("manual.{drop,keep}", true)`.

**Pull request merged**